### PR TITLE
modularize, expand documentation, expand curve list

### PR DESCRIPTION
**Pull request by calvinmetcalf**

### CODE REVIEW

**Comment by dcousens** [comment]
Will review. ETA 24 - 48 hrs.

**Comment by calvinmetcalf** [comment]
thanks! no hurry though we've modularized all the popular ones.

**Comment by calvinmetcalf** [comment]
ping

File: test.js [hunk 1]
```diff
@@ -59,6 +62,6 @@ function run(i) {
 
 
 var i = 0;
-while (++i < 10) {
```
Comment (dcousens) [inline test.js:62]: why `100`? arbitrary?

File: index.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-exports.__browserify = require('./inject');
+var createECDH = require('crypto').createECDH;
 
-exports.__browserify(require('crypto'), exports);
\ No newline at end of file
+module.exports = createECDH || require('./browser');
```
Comment (dcousens) [inline index.js:3]: LGTM, haven't tested though.

File: browser.js [hunk 3]
```diff
@@ -56,11 +88,16 @@ ECDH.prototype.setPrivateKey = function (priv, enc) {
 	_priv = _priv.toString(16);
 	this.keys._importPrivate(_priv);
 };
-function returnValue(bn, enc) {
+function returnValue(bn, enc, len) {
```
Comment (dcousens) [inline browser.js]: `returnValue` is a bit lame.  It looks like this code is padding to a given length and then returning the buffer in a particular encoding.

Can we make this into 2 separate steps?

File: browser.js [hunk 4]
```diff
@@ -1,10 +1,42 @@
...
+	ed25519: {
+		name: 'ed25519',
+		byteLength: 32
+	}
+};
+aliases.p224 = aliases.secp224r1;
+aliases.p256 = aliases.secp256r1 = aliases.prime256v1;
+aliases.p192 = aliases.secp192r1 = aliases.prime192v1;
+function ECDH(curve) {
+	if (!(this instanceof ECDH)) {
```
Comment (dcousens) [inline browser.js]: Can we do this [consistently](https://github.com/crypto-browserify/createHmac/pull/1/files#diff-de1a5a7de991855714cdf433e9f8be96R9) across modules?

File: browser.js (hunk 3 above)
Comment (dcousens) [inline browser.js]: If you ACK, I'll go ahead and add the commits.

File: test.js (hunk 1 above)
Comment (calvinmetcalf) [inline test.js:62]: there was an intermittent error and when i set it to 100 I could reliably get that error

File: browser.js (hunk 4 above)
Comment (calvinmetcalf) [inline browser.js]: so I switched it in Hmac becasue it was a huge pain to always have 2 functions with the same name in the stack so I switched Hmac to have a factory that calls new

Comment (calvinmetcalf) [inline browser.js]: so yes I _ususally_ agree with you that this is preferred when constructors are available but I don't see it as a substitute for a factory.

**Comment by calvinmetcalf** [comment]
updated

**Comment by ForbesLindesay** [comment]
@dcousens can this be merged now?

**Comment by calvinmetcalf** [comment]
I [updated the .travis.yml](https://github.com/crypto-browserify/createECDH/commit/5576ed5b70f622f6a70884ab963334d46e5a8afb) if that passes I'll publish.

**Pull request merged**